### PR TITLE
feat: [CDS-42071]: ECS Max Desired Rollback count

### PR DESCRIPTION
**Pull request by akhilpandey1**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38802)

### CODE REVIEW

**Comment by akhilpandey1** [comment]
trigger feature-build

File: 930-delegate-tasks/src/main/java/software/wings/delegatetasks/aws/ecs/ecstaskhandler/deploy/EcsDeployCommandHandler.java [hunk 1]
```diff
@@ -155,6 +169,40 @@ public EcsCommandExecutionResponse executeTaskInternal(EcsCommandRequest ecsComm
...
+      Service lastSuccessfulService =
+          awsHelperService
+              .describeServices(resizeParams.getRegion(), ecsCommandRequest.getAwsConfig(), encryptedDataDetails,
+                  new DescribeServicesRequest()
+                      .withCluster(resizeParams.getClusterName())
+                      .withServices(newInstanceDataList.get(0).getName()))
+              .getServices()
+              .get(0);
+      int unsucessfulServiceCount =
+          Integer.max(unsucessfulService.getDesiredCount(), unsucessfulService.getRunningCount());
```
Comment (akhilpandey1) [inline 930-delegate-tasks/src/main/java/software/wings/delegatetasks/aws/ecs/ecstaskhandler/deploy/EcsDeployCommandHandler.java:196]: Remove running count from computation

File: 930-delegate-tasks/src/main/java/software/wings/delegatetasks/aws/ecs/ecstaskhandler/deploy/EcsDeployCommandHandler.java [hunk 2]
```diff
@@ -91,6 +95,16 @@ public EcsCommandExecutionResponse executeTaskInternal(EcsCommandRequest ecsComm
 
         newInstanceDataList = resizeParams.getNewInstanceData();
         oldInstanceDataList = resizeParams.getOldInstanceData();
+        int desiredRollbackCount;
+        if (resizeParams.isFFMaxDesiredCountEnabled()) {
+          desiredRollbackCount = getECSMaxDesiredRollbackCount(
+              ecsCommandRequest, encryptedDataDetails, resizeParams, newInstanceDataList, oldInstanceDataList);
+          desiredRollbackCount = Integer.max(newInstanceDataList.get(0).getDesiredCount(), desiredRollbackCount);
```
Comment (akhilpandey1) [inline 930-delegate-tasks/src/main/java/software/wings/delegatetasks/aws/ecs/ecstaskhandler/deploy/EcsDeployCommandHandler.java:102]: This is redundant

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]

**Comment by akhilpandey1** [comment]
trigger pmd,trigger checkstyle

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> Reviewed 4 of 5 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38802)* status: :shipit: complete! ✓

**Review by tchaurasiya — APPROVED** [review]
> Reviewed 3 of 5 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38802)* status: :shipit: complete! ✓

**Pull request merged**